scripts/publishMaven.js

```diff
@@ -0,0 +1,270 @@
+}
+
+ /**
+ * Task promote: Promote a repo to get it picked up by Maven Central.
+ * 
+ * Required binaries:
+ * - curl
+ * 
+ * Required Environment Variables:
+ * - NEXUS_OSSRHUSER: username.
+ * - NEXUS_OSSRHPASS: password.
+ * - NEXUS_STAGINGPROFILEID: identifier of the repo to promote.
+ * - NEXUS_STAGINGREPOID: id of staging repo with artifacts to promote.
+ */
+function promoteToCentral(configs) {
+    let message = "";
+    console.log("\n========Nexus: Promote=======");
+    try {
+        console.log(`Starting to promote staging repository ${configs.stagingRepoId} ...`);
+        console.log(`curl -i -X POST -d "<promoteRequest><data><stagedRepositoryId>${configs.stagingRepoId}</stagedRepositoryId></data></promoteRequest>" -H "Content-Type: application/xml" -u **:** -k https://oss.sonatype.org/service/local/staging/profiles/${configs.nexus_stagingProfileId}/promote`);
+        message = childProcess.execSync(`curl -i -X POST -d "<promoteRequest><data><stagedRepositoryId>${configs.stagingRepoId}</stagedRepositoryId></data></promoteRequest>" -H "Content-Type: application/xml" -u ${configs.nexus_ossrhuser}:${configs.nexus_ossrhpass} -k https://oss.sonatype.org/service/local/staging/profiles/${configs.nexus_stagingProfileId}/promote`);
+        message = message.toString();
+        console.log(message);
+    } catch (ex) {
+        console.error("\n\n[Failure] Promoting staging repository failed.");
+        console.error(!message ? ex : message.toString());
+        process.exit(1);
+    }
+    const success = isReleased(configs);
+    console.log("Below is the public repository url, you could manually validate it.");
+    console.log(`https://oss.sonatype.org/content/groups/public/${configs.groupId.replace(/\./g, "/")}`);
+    console.log("\n\n");
+    if (success) {
+        console.log("\n\n[Success] Nexus: Promote completion.");
+    } else {
+        console.error("\n\n[Failure] Nexus: Promote failed.");
+        process.exit(1)
+    }
+}
+
+function isReleased(configs) {
+    let pollingCount = 0;
+    const MAX_POLLINGS = 10;
+    for (; pollingCount < MAX_POLLINGS; pollingCount++) {
+        console.log(`\nPolling the release operation finished or not...`);
+        console.log(`curl -X GET -H "Content-Type:application/xml" -u **:** -k https://oss.sonatype.org/service/local/staging/repository/${configs.stagingRepoId}`);
+        message = childProcess.execSync(`curl -X GET -H "Content-Type:application/xml" -u ${configs.nexus_ossrhuser}:${configs.nexus_ossrhpass} -k https://oss.sonatype.org/service/local/staging/repository/${configs.stagingRepoId}`);
+        const status = extractStatus(message.toString());
+        console.log(status);
+        if (status !== "closed") {
+            return true;
+        }
+        // use system sleep command to pause the program.
+        childProcess.execSync(`sleep 6s`);
+    }
+    return false;
+}
+
+function checkPrerequisite(configs) {
+    if (!configs.releaseVersion) {
```

Comment:
Null check for other necessary keys either?

Reply:
will add artifactFolder and user/pass etc.